### DISCUSSION TITLE
Why does the warning occur in the server console "WARNING: Running slowly (52 updates/s)!" ?

### DISCUSSION THREAD

**Discussion by pavel4win** [FakeFishGames/Barotrauma#13953] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

Although The server configuration is sufficient.
Often a warning appears in the console
This happens quite often
This happens on a rented virtual server
2 cores 3.3 GHz, 3 GB. RAM

This happens on a home dedicated server
4 cores 3.3 GHz, 16 GB RAM

All client's characters at this time, begin to move jerkily.

### Reproduction steps

I don't know how to reproduce the error. But it occurs every evening and is very annoying.

### Bug prevalence

Happens regularly

### Single player or multiplayer?

Multiplayer hosted using a dedicated server

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by pavel4win** [comment]
I again faced the error "WARNING: Running slowly (** updates/s)!"
I am host the dedic every weekend
Can I help to devs to collect more info about the error anyway?

**Reply by SomeRandomNoobKekeke** [reply]
Isn't those just normal lag spikes?

**Reply by pavel4win** [reply]
I don't know. That's why I'm asking. 
So this warning appears because of slow internet and not because of a slow computer? 
Then it's not surprising, because 50 updates per second will be with a ping of 20. 
And ping is usually much higher, about 50-60.

**Reply by SomeRandomNoobKekeke** [reply]
https://github.com/FakeFishGames/Barotrauma/blob/7d5b7a310a52372dac0a8bac90fd2a6a2ab342dc/Barotrauma/BarotraumaServer/ServerSource/GameMain.cs#L374
![Definitely a screenshot](https://github.com/user-attachments/assets/78013e78-f57b-4d89-9e30-555d7100c93d)
No, it's means 4 cores 3.3 GHz, 16 GB RAM is not enough for barotrauma
Barotrauma is known for it's mind-bending performance, i'm surprised you ever had 60fps on 4 cores 3.3 GHz
![Screenshot_9](https://github.com/user-attachments/assets/37127a11-a317-404f-a9af-ddab4244d827)
![Screenshot_10](https://github.com/user-attachments/assets/b94c1de7-deda-457f-acf5-8077c0ce35d3)

**Reply by pavel4win** [reply]
The warning goes from the dedicated game server that does not have a graphical interface at all.
What FPS are we talking about?
Anyway I asked a question about multi-core earlier.
In response, the developers wrote to me that barotrauma does not care how many cores the processor has, the game uses only one core. So it must be very powerful core.

The problem is that the server is currently hosted on a cloud VPS
There I cannot change the processor speed, I can only add the number of cores.
After all, I understand from the above, in order for the warning to disappear, I have to install the server on a more powerful computer.

**Reply by Regalis11** [reply]
> In response, the developers wrote to me that barotrauma does not care how many cores the processor has, the game uses only one core. So it must be very powerful core.

I'm not sure which response you're referring to, but this is not entirely accurate: the game does use multiple cores, but it's true that most of the "heavy lifting" is done in one thread, and as such single-core performance tends to affect the performance of the game a lot.

But anyhow, this warning means that the server's having trouble running the game logic at the normal fixed rate of 60 updates per frame - it's unrelated to the internet connection or the connection with the clients, and just means the computer is having trouble running the game. It's hard to say what exactly is causing this to happen in your case, but mods or unusually performance-heavy situations may cause it to occur.

**Reply by pavel4win** [reply]
I don't remember exactly where and from whom I got this answer, maybe it was in Discord. I have got an answer that a multi-core server only makes sense because a separate core will service the system needs.
My barotrauma dedicated linux server not have mods at all. Pure vanilla.
I think the most obvious reason for this warning is that the server is running on a VPS and the host services not only my server.
On next weekend I will try add more cores.
I realized that 2 processor core on a VPS actually means that the host will allocate 2 times more processor time to my server. So it should work better no matter whether the barotrauma is multithreaded.
